### PR TITLE
Set callback URL of fragment apps to app owner org's commonauth endpoint

### PR DESCRIPTION
**Pull request by piyarathnalakmali**
## Purpose
$subject

Depends on 
- https://github.com/wso2/carbon-identity-framework/pull/4372
- https://github.com/wso2-extensions/identity-organization-management-core/pull/48
- https://github.com/wso2-enterprise/asgardeo-deployment-helm-is/pull/619

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 1]
```diff
@@ -582,6 +580,20 @@ private OAuthConsumerAppDTO createOAuthApplication(String mainAppName)
         return getOAuthAdminService().registerAndRetrieveOAuthApplicationData(consumerApp);
     }
 
+    private String getCallbackURL(String ownerOrgId) throws URLBuilderException, OrganizationManagementException {
+
+        String tenantDomain = getOrganizationManager().resolveTenantDomain(ownerOrgId);
+        ServiceURL commonAuthServiceUrl;
+        if (useTenantQualifiedURLs(ownerOrgId)) {
```
[the remark this answers is not in the document]
Comment (piyarathnalakmali) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java]: Addressed with df1452d87980ca0433094bc45443a067eee0e8f9

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 2]
```diff
@@ -582,6 +580,20 @@ private OAuthConsumerAppDTO createOAuthApplication(String mainAppName)
...
+            commonAuthServiceUrl = ServiceURLBuilder.create().addPath(FrameworkConstants.COMMONAUTH)
+                    .setTenant(tenantDomain).build();
+        } else {
+            commonAuthServiceUrl = ServiceURLBuilder.create().addPath(FrameworkConstants.COMMONAUTH)
+                    .setTenant(tenantDomain).setOrganization(ownerOrgId).build();
+        }
+        return commonAuthServiceUrl.getAbsolutePublicURL();
```
[the remark this answers is not in the document]
Comment (piyarathnalakmali) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java]: Addressed with df1452d87980ca0433094bc45443a067eee0e8f9

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 3]
```diff
@@ -538,7 +538,8 @@ public void shareApplication(String ownerOrgId, String sharedOrgId, ServiceProvi
             // Create Oauth consumer app to redirect login to shared (fragment) application.
             OAuthConsumerAppDTO createdOAuthApp;
             try {
-                createdOAuthApp = createOAuthApplication(mainApplication.getApplicationName());
+                String callbackUrl = getCallbackURL(ownerOrgId);
```
[the remark this answers is not in the document]
Comment (piyarathnalakmali) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java]: Addressed with df1452d87980ca0433094bc45443a067eee0e8f9

**Review by AnuradhaSK — DISMISSED** [review]

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 4]
```diff
@@ -582,6 +579,18 @@ private OAuthConsumerAppDTO createOAuthApplication(String mainAppName)
         return getOAuthAdminService().registerAndRetrieveOAuthApplicationData(consumerApp);
     }
 
+    private String resolveCallbackURL(String ownerOrgId) throws URLBuilderException, OrganizationManagementException {
+
+        String tenantDomain = getOrganizationManager().resolveTenantDomain(ownerOrgId);
+        ServiceURLBuilder commonAuthServiceUrl = ServiceURLBuilder.create().addPath(FrameworkConstants.COMMONAUTH)
+                .setTenant(tenantDomain);
+        if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain) &&
+                isOrganizationQualifiedURLsSupported(ownerOrgId)) {
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java:588]:
```suggestion
                !isOrganizationQualifiedURLsDisabled(ownerOrgId)) {
```
We can refactor the method only if we are fine to resolve the suggestions for this PR.
https://github.com/wso2-extensions/identity-organization-management-core/pull/48

**Review by sadilchamishka — APPROVED** [review]

**Review by sadilchamishka — APPROVED** [review]

**Pull request merged**